src/hooks.js

```diff
@@ -60,13 +60,13 @@ export function useSteps(steps) {
     if (step < steps - 1) {
       setStep(step + 1)
     }
-  }, [step, steps])
+  }, [setStep, step, steps])
 
   const prev = useCallback(() => {
     if (step > 0) {
       setStep(step - 1)
     }
-  }, [step, steps])
+  }, [setStep, step])
 
   return {
```

Comment:
I found two ways to solve this: by using a reducer, or by using a reference. It seems [the reducer solution](https://overreacted.io/a-complete-guide-to-useeffect/#decoupling-updates-from-actions) is the most react-y way of doing it, [so I opened a PR doing this](https://github.com/aragon/aragon/pull/774).

Re-reading this article after having played a bit with hooks was actually an eye opener about `useReducer()`: I was trying to avoid it until I had a case for something looking like a redux store, but I understand now how it can be useful for much smaller things, like in that case.

One thing I don’t really like is the way the `steps` is passed: the article recommends to [declare the reducer in the render function](https://overreacted.io/a-complete-guide-to-useeffect/#why-usereducer-is-the-cheat-mode-of-hooks) to make it aware of props, but it also says “this pattern disables a few optimizations so try not to use it everywhere, but you can totally access props from a reducer if you need to”. I can imagine that the `useReducer` hook would have to update its reducer function every time we render, which might not be optimal depending on how it’s done internally. That’s why I opted for passing it in the action directly, which seems acceptable for such a small case.

Reply:
> This is why I like to think of useReducer as the “cheat mode” of Hooks. It lets me decouple the update logic from describing what happened. This, in turn, helps me remove unnecessary dependencies from my effects and avoid re-running them more often than necessary.

This sounds like exactly what we were encountering :smile: